test/evergreen/build_windows.ps1

```diff
@@ -1,9 +1,40 @@
 param (
     [bool]$configure = $false,
     [bool]$build = $false,
-    [string]$vcvars_bat = "C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Auxiliary\Build\vcvars64.bat"
+    [string]$vcvars_bat = $null
 )
 
+function Find-VcVars {
+    $vswhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (-not (Test-Path $vswhere)) {
+        Write-Warning "vswhere.exe not found at $vswhere"
+        return $null
+    }
```

Comment:
If this PR blocks something, then probably this comment is not so important since it is debatable, but I am thinking about handling all the failures in this script consistently.

Currently this script contains `Die-On-Failure` function that does throw in case of non-zero error code and doesn't print any error messages. This patch brings another logic that does `exit 1` in case of error and prints additional errors/warnings. I think it is good to handle all the failures similarly (like extend `Die-On-Failure` by message argument) but that's not required.

Let me know what you think.

Reply:
These are all good points!
I'd just use `throw` directly since I don't want to check last error code.